grouper/fe/handlers.py

```diff
@@ -858,6 +886,172 @@ def get(self, group_id=None, name=None):
         )
 
 
+class GroupPermissionRequest(GrouperHandler):
+    def _prep_form(self, form, args_by_perm):
+        form.permission_name.choices = [("", "")] + sorted([(p, p) for p in args_by_perm.keys()])
+        form.argument_select.choices = [("", "")]
+
+        return form
+
+    def _prep_form_for_display(self, form):
+        # hack: we don't want validation to puke if one is undefined but we
```

Comment:
this form allows selecting an argument either from a dropbown or a free form text field, depending on what grants are available. so one of these fields should be undefined/empty. is there a better way to express this mutual exclusivity within wtform itself?